### ISSUE TITLE
Surface the joystick name as reported by the hardware.

### ISSUE THREAD

**Issue by blakeohare** [opentk/opentk#312]
Currently the only string identifier available for joystick devices is the Description property, which is just "Joystick/Joystick # (# Axes, # Buttons)". This isn't terribly useful to uniquely identify multiple joysticks in, say, an input config menu, and is name-collision prone for different devices.

**Comment by emacinnes** [comment]
This is definitely needed, not only for binding a specific controller config (e..g an x52 w. flight pedals) but also for telling the Joysticks and GamePads apart, as the Gamepads get repeated in the Joystick class.

I'd hasten to add button names as well, rather than just button ID's if at all possible to improve the UI binding experience for users, as buttons are labelled as X A, Y, B, (i) and so on, and not button 0, button 1.

**Comment by varon** [comment]
This would be nice to have. If someone is able to PR this in, we would welcome the contribution.

At this time, however, we don't have the manpower to undertake this.

**Comment by Frassle** [comment]
#651 to resolve this

**Comment by varon** [comment]
@frederikja163  solved by the HID input PR?

**Comment by frederikja163** [comment]
Yup, as of opentk 4.0 they will be identifiable in two different ways, and ID (which should be consistent with their position in the joystickstates array) and their name, which is a string

TL;DR the HID input did solve this for 4.0

**Comment by varon** [comment]
Closed by 4.x
